text/001x-battery_improvements.md

```diff
@@ -0,0 +1,293 @@
+          Other faults like MAV_BATTERY_STATUS_FLAGS_FAULT_OVER_VOLT may also be set, indicating the cause of the protection fault.
+        </description>
+      </entry>
+      <entry value="512" name="MAV_BATTERY_STATUS_FLAGS_FAULT_OVER_VOLT">
+        <description>One or more cells are above their maximum voltage rating.</description>
+      </entry>
+      <entry value="1024" name="MAV_BATTERY_STATUS_FLAGS_FAULT_UNDER_VOLT">
+        <description>
+          One or more cells are below their minimum voltage rating.
+          A battery that had deep-discharged might be irrepairably damaged, and set both MAV_BATTERY_STATUS_FLAGS_FAULT_UNDER_VOLT and MAV_BATTERY_STATUS_FLAGS_BAD_BATTERY.
+        </description>
+      </entry>
+      <entry value="2048" name="MAV_BATTERY_STATUS_FLAGS_FAULT_OVER_TEMPERATURE">
+        <description>Over-temperature fault.</description>
+      </entry>
+      <entry value="4096" name="MAV_BATTERY_STATUS_FLAGS_FAULT_UNDER_TEMPERATURE">
+        <description>Under-temperature fault.</description>
+      </entry>
+      <entry value="8192" name="MAV_BATTERY_STATUS_FLAGS_FAULT_OVER_CURRENT">
+        <description>Over-current fault.</description>
+      </entry>
+      <entry value="16384" name="MAV_BATTERY_STATUS_FLAGS_FAULT_SHORT_CIRCUIT">
+        <description>
+          Short circuit event detected.
+          The battery may or may not be safe to use (check other flags).
+        </description>
+      </entry>
+      <entry value="32768" name="MAV_BATTERY_STATUS_FLAGS_FAULT_INCOMPATIBLE_VOLTAGE">
+        <description>Voltage not compatible with power rail voltage (batteries on same power rail should have similar voltage).</description>
+      </entry>
+      <entry value="65536" name="MAV_BATTERY_STATUS_FLAGS_FAULT_INCOMPATIBLE_FIRMWARE">
+        <description>Battery firmware is not compatible with current autopilot firmware.</description>
+      </entry>
+      <entry value="131072" name="MAV_BATTERY_STATUS_FLAGS_FAULT_INCOMPATIBLE_CELLS_CONFIGURATION">
+        <description>Battery is not compatible due to cell configuration (e.g. 5s1p when vehicle requires 6s).</description>
+      </entry>
+      <entry value="262144" name="MAV_BATTERY_STATUS_FLAGS_CAPACITY_RELATIVE_TO_FULL">
+        <description>
+          Battery capacity_consumed and capacity_remaining values are relative to a full battery (they sum to the total capacity of the battery).
+          This flag would be set for a smart battery that can accurately determine its remaining charge across vehicle reboots and discharge/recharge cycles.
+          If unset the capacity_consumed indicates the consumption since vehicle power-on, as measured using a power monitor. The capacity_remaining, if provided, indicates the estimated remaining capacity on the assumption that the battery was full on vehicle boot.
+          If unset a GCS is recommended to advise that users fully charge the battery on power on.
```

Comment:
AP has this command `MAV_CMD_BATTERY_RESET ` to indicate less than 100% charge on boot up. Very useful with fuel systems. 

https://mavlink.io/en/messages/ardupilotmega.html#MAV_CMD_BATTERY_RESET

```suggestion
          If unset a GCS is recommended to advise that users fully charge the battery on power on. Or indicate the level of charge of the via another manner.
```

Reply:
MAV_CMD_BATTERY_RESET conceptually sounds like a very good idea. Smart battery users won't care, but potentially very handy for the power monitor case. Not sure I like the bitmask approach though.

Reply:
It is most handy for those using petrol engines. As it is common to only fill them part way. 

On the bitmask I think it just allows you to reset multiple batteries at once.

Reply:
Thanks. I understand the bitmask. It's fine provided you only have 8 batteries - so my concern was "not scalable". Probably fine for the vast majority of systems for now.

I'm a little concerned about a message for battery resetting where the main use case is petrol engines :-).

I'll attempt a PR into common and see if there is interest in PX4 in taking it, and/or possible modifications.

Reply:
> Thanks. I understand the bitmask. It's fine provided you only have 8 batteries - so my concern was "not scalable". Probably fine for the vast majority of systems for now.
> 
> I'm a little concerned about a message for battery resetting where the main use case is petrol engines :-).
> 
> I'll attempt a PR into common and see if there is interest in PX4 in taking it, and/or possible modifications.

I'm not sure if the main purpose is for petrol engines just the one fresh on my mind among those I'm assisting. 

I agree the bitmask approach is limiting.

Any ways I think this is this nearly there so I'm going to try to bring this up in tonight's devcall so we can get feedback from the AP side besides me :)

Reply:
Thanks! Appreciate it!